### PR TITLE
Criticality dev

### PR DESCRIPTION
**Pull request by Stern5497**
### Criticality Prediction:

**What changed?**
1. Add multi class label citation-label
2. Enable return of multible labels in dataset creator
3. Create reports and save in one place
4. Enable return of multible labels in dataset creator
5. Add save_reports for criticality_dataset
6. Add secret test
7. Add num_tokens per feature col
8. Save list of labels in different files
9. Save feature cols to hugging face

**Why were there changes?**
In one dataset multiple feature cols like facts, considerations, judgments, ... get treated at once. Many methods needed to be adapted to handle multiple feature cols. As an example we need num_nokens for each or when creating reports do it per label.

When we create the criticality dataset we define bge-lable and citation-label. Multiple label handling needs to be adapted. In case of citation-label we need to be able to handle multi-class labels.

**Important**
The amount of citations which define one class of ciatation_label can be set in init function. It doesn't matter how many different classes one chooses.

It is only possible at the moment to chose 1 or 2 feature_cols. Anything else will produce an error.

### CODE REVIEW

File: scrc/data_classes/ruling_citation.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def __init__(self, citation_str, language):
             raise ValueError(f"The Citation String ({citation_str}) could not be parsed successfully.")
 
     def __str__(self):
-        return f"{self.ruling_str} {self.year} {self.volume} {self.page_number}"
+        return f"{self.ruling_str}-{self.year}-{self.volume}-{self.page_number}"
```
Comment (JoelNiklaus) [inline scrc/data_classes/ruling_citation.py:33]: What is the reason for this?
I think usually it is without dashes: e.g. BGE 132 II 257

File: scrc/dataset_creation/dataset_creator.py [hunk 2]
```diff
@@ -198,14 +198,18 @@ def save_huggingface_dataset(self, splits, feature_col_folder):
         :param feature_col_folder:      name of folder
         """
         huggingface_dir = self.create_dir(feature_col_folder, 'huggingface')
-        for split in ['train', 'val', 'test']:
+        for split in ['train', 'val', 'test', 'secret_test']:
```
Comment (JoelNiklaus) [inline scrc/dataset_creation/dataset_creator.py:201]: Nice :)

File: scrc/utils/sql_select_utils.py [hunk 3]
```diff
@@ -408,10 +411,6 @@ def get_region(canton):
         if np.isnan(canton):
             return None
         canton = Canton(canton)
-    if isinstance(canton, str):
-        for canton_enum in Canton:
-            if canton_enum.name == canton:
-                canton = canton_enum
```
Comment (JoelNiklaus) [inline scrc/utils/sql_select_utils.py:414]: Is there a reason for removing this?

**Review by JoelNiklaus — COMMENTED** [review]
It looks good already! 

Great description of the changes.
Would it be possible to add some a bit more detailed method comments so it is easier to understand what is going on? (especially in the criticality_dataset_creator and the dataset_creator)

File: scrc/data_classes/ruling_citation.py (hunk 1 above)
Comment (Stern5497) [reply scrc/data_classes/ruling_citation.py:33]: The reason is that I want to compare this string to given bge file_number. Those file_numbers are stored in our db with dashes.

File: scrc/utils/sql_select_utils.py (hunk 3 above)
Comment (Stern5497) [reply scrc/utils/sql_select_utils.py:414]: No there is not, it was not removed on purpose. I will include it again.

**Comment by Stern5497** [comment]
**Additional Changes:**
- Add possibility to count citations per case only once -> otherwhise it depends for example a author how often he uses citations in one case.
- Bug detected: To create sub-datasets we still need to implement handling of multiple feature-cols per dataframe.

**Comment by JoelNiklaus** [comment]
Thank you very much!

**Pull request merged**